### PR TITLE
PLAT-145554: Fix to focus elements at scroll boundaries by hover when `hoverToScroll` is `true`

### PR DESCRIPTION
**Pull request by 0x64**
Enact-DCO-1.0-Signed-off-by: Seungcheon Baek (sc.baek@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Elements on scroll edges don't get focus by hover when `hoverToScroll` is `true`.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
The cause of this bug is that a DOM node to catch hovering blocks mouse events on elements.
This PR shows and hides the DOM node along with scroll status.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-145554
enactjs/enact/pull/2930

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ The following is a curated list of changes in the Enact sandstone module, newest
 
 - `sandstone/Picker` value to not marquee when changing `title`
 - `sandstone/Scroller` and `sandstone/VirtualList` to scroll by hover when scrollbar is hidden
+- `sandstone/Scroller` and `sandstone/VirtualList` to focus elements on scroll edges when `hoverToScroll` is `true`
```
Comment (MikyungKim) [inline CHANGELOG.md:11]:
```suggestion
- `sandstone/Scroller` and `sandstone/VirtualList` to focus elements on scroll start and end positions  when `hoverToScroll` is `true`
```

File: useScroll/HoverToScroll.js [hunk 2]
```diff
@@ -47,30 +49,29 @@ const directionToFocus = {
...
-
-	const [enable, setEnable] = useState();
-
-	useLayoutEffect(() => {
-		if (scrollContainer) {
-			const {[getBoundsPropertyNames(direction).canScroll]: canScroll, getScrollBounds} = scrollContainer;
-			setEnable(canScroll && getScrollBounds ? canScroll(getScrollBounds()) : null);
-		}
-	}, [direction, enable, scrollContainer]);
+	mutableRef.current.scrollContainer = current;
```
Comment (MikyungKim) [inline useScroll/HoverToScroll.js:69]: Let's not do this...

File: CHANGELOG.md (hunk 1 above)
Comment (0x64) [reply CHANGELOG.md]: I updated from 'on scroll edges' to 'at scroll boundaries'.

File: useScroll/HoverToScroll.js (hunk 2 above)
Comment (0x64) [reply useScroll/HoverToScroll.js]: 🆗

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**